exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/DrillFileSystem.java

```diff
@@ -89,22 +89,36 @@ public DrillFileSystem(Configuration fsConf) throws IOException {
   }
 
   public DrillFileSystem(Configuration fsConf, OperatorStats operatorStats) throws IOException {
-    this.underlyingFs = FileSystem.get(fsConf);
+    this(fsConf, URI.create(fsConf.getRaw(FS_DEFAULT_NAME_KEY)), operatorStats);
```

Comment:
Actually, why do we need to do this step? The original code seems to do the right thing:

```
class FileSystem ...
  public static FileSystem get(Configuration conf) throws IOException {
    return get(getDefaultUri(conf), conf);
  }
  public static URI getDefaultUri(Configuration conf) {
    return URI.create(fixName(conf.get(FS_DEFAULT_NAME_KEY, DEFAULT_FS)));
  }
```

That is, the original code gets the file system using the URI stored in the config. Standard practice is that the caller must have set the file system property: that is how we tell a "file:///" system from an HDFS system, etc.

So, isn't the problem here with the caller?

Reply:
Removed the other constructor and am creating underlying filesystem directly